common/audit.cto

```diff
@@ -123,3 +123,21 @@ transaction ClauseTriggered extends ClauseTriggerEvent {
   o String stack optional  // Required if severity == ERROR
   o Boolean test default=false optional
 }
+
+abstract transaction FlowStep {
```

Comment:
Not sure you want these to be abstract, do you? I'm assuming you will be creating instances of this type - not just sub-classes?

Reply:
That's correct. I just copied from ClauseTriggered, as the use of FlowError should exactly mirror that of ClauseTriggered. Should both be without abstract, perhaps?

Reply:
Sounds like yes, if we are expecting that $class to be in the JSON audit trail.

Reply:
It sounds like validation for the JSON must be off, or we would have seen an exception when we tried to deserialise an abstract type.